src/pool/srv_pool.c

```diff
@@ -6829,10 +6829,10 @@ pool_svc_update_map(struct pool_svc *svc, crt_opcode_t opc, bool exclude_rank,
 
 	if (svc->ps_pool->sp_reint_mode == DAOS_REINT_MODE_NO_DATA_SYNC) {
 		D_DEBUG(DB_MD, "self healing is disabled for no_data_sync reintegration mode.\n");
-		if (opc == POOL_EXCLUDE || opc == POOL_DRAIN) {
-			rc = ds_pool_tgt_exclude_out(svc->ps_pool->sp_uuid, &target_list);
+		if (opc == MAP_REINT) {
```

Comment:
when exclude, it still will trigger rebuild?

Reply:
No, rebuild will be skipped.

Reply:
where is the code to skip it? I mean the rebuild for exclude. user first exclude, then reint right?

Reply:
sp_reint_mode is no_data_sync, then nothing is done, and followng D_GOTO(out, rc) will be called? which skipped rebuild, am i missing something?

Reply:
To avoid the problem, probably can set the tgt as DOWN when EXCLUDE, to avoid always read from it.
but later when REINT, cannot directly call ds_pool_tgt_finish_rebuild(), may define a new map_update_opc MAP_DOWN2UPIN specially for no_data_sync usage to update pool map.
How do you think?

Reply:
You are right. thanks for good suggestions!

Reply:
> To avoid the problem, probably can set the tgt as DOWN when EXCLUDE, to avoid always read from it. but later when REINT, cannot directly call ds_pool_tgt_finish_rebuild(), may define a new map_update_opc MAP_DOWN2UPIN specially for no_data_sync usage to update pool map. How do you think?

For reintegration, in pool_svc_update_map(), pool_svc_update_map_internal() is called first which will mark targets from DOWN to UP firstly(this is a bit like delay rebuild case, co_flags will set DOWN2UP). and then here we call ds_pool_tgt_finish_rebuild() which will mark targets from UP to UPIN. Looks simple for me.